experiments/conductor/cmd/runner/crd_generator_commands.go

```diff
@@ -0,0 +1,320 @@
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := filepath.Join(opts.branchRepoDir)
+
+	var out strings.Builder
+	checkoutBranch(branch, workDir, &out)
+
+	// Create the apis/<service>/<version> directory
+	serviceDir := filepath.Join(workDir, "apis", branch.Group, "v1alpha1")
+	if err := os.MkdirAll(serviceDir, 0755); err != nil {
+		log.Fatal(err)
+	}
+
+	// Create or update generate.sh
+	scriptPath := filepath.Join(serviceDir, "generate.sh")
+	// Check if generate.sh already exists.
+	if _, err := os.Stat(scriptPath); errors.Is(err, os.ErrNotExist) {
+		// File doesn't exist, use template approach
+		log.Printf("Creating new generate.sh at %s", scriptPath)
+
+		// Replace template markers with actual values and write to file
+		writeTemplateToFile(branch, scriptPath, GENERATOR_SCRIPT_TEMPLATE)
+	} else {
+		// File exists, use codebot to update it
+		log.Printf("Updating existing generate.sh at %s", scriptPath)
+
+		// Delete then write the prompt file.
+		promptPath := filepath.Join(workDir, "mockgcp", "crdgen_prompt.txt")
+		writeTemplateToFile(branch, promptPath, UPDATE_GENERATE_SCRIPT_PROMPT)
+
+		cfg := CommandConfig{
+			Name:    "CODEBOT GENERATE",
+			Cmd:     "codebot",
+			Args:    []string{"--ui-type=prompt", "--prompt=mockgcp/crdgen_prompt.txt"},
+			WorkDir: workDir,
+		}
+		if err := executeCommand(cfg, &out); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	// Add and commit changes
+	scriptRelativePath := fmt.Sprintf("apis/%s/v1alpha1/generate.sh", branch.Group)
+	gitAdd(workDir, &out, scriptRelativePath)
+	gitCommit(workDir, &out, fmt.Sprintf("add/update crd generation script for %s", branch.Group))
+
+	// Run the generator script
+	cfg := CommandConfig{
+		Name:    "Generator script",
+		Cmd:     scriptPath,
+		WorkDir: workDir,
+	}
+	if err := executeCommand(cfg, &out); err != nil {
+		log.Fatal(err)
+	}
+
+	// Add and commit generated files
+	gitAdd(workDir, &out,
+		fmt.Sprintf("apis/%s/v1alpha1/", branch.Group),
+		fmt.Sprintf("pkg/controller/direct/%s/", branch.Group),
+		"config/crds/resources/")
+	gitCommit(workDir, &out, fmt.Sprintf("autogenerated types and CRDs using %s", scriptRelativePath))
+}
+
+func generateTypesAndMapper(opts *RunnerOptions, branch Branch) {
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := filepath.Join(opts.branchRepoDir)
+
+	var out strings.Builder
+	checkoutBranch(branch, workDir, &out)
+
+	// Change to controllerbuilder directory
+	controllerBuilderDir := filepath.Join(workDir, "dev/tools/controllerbuilder")
+
+	// Generate types
+	cfg := CommandConfig{
+		Name: "Generate types",
+		Cmd:  "go",
+		Args: []string{
+			"run", ".",
+			"generate-types",
+			"--service", branch.Package,
+			"--api-version", fmt.Sprintf("%s.cnrm.cloud.google.com/v1alpha1", branch.Group),
+			"--resource", fmt.Sprintf("%s:%s", branch.Kind, branch.Proto),
+		},
+		WorkDir: controllerBuilderDir,
+	}
+	if err := executeCommand(cfg, &out); err != nil {
+		log.Fatal(err)
```

Comment:
Please make sure to log the error & out just in case before calling fatal.

Reply:
executeCommand logs it.